### PR TITLE
Populate graphql with transaction data

### LINKED ISSUE #1284: Include the decoded payload fields into GraphQL
**Describe what you want implemented** 
GraphQL queries need to be reviewed in order to expose the fields to the block explorer

**Describe "Why" this is needed**
TBD

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1284

### CODE REVIEW

**Review by herr-seppia — CHANGES_REQUESTED** [review]
RFC in reward calculation

File: pkg/gql/query/blocks.go [hunk 1]
```diff
@@ -64,22 +67,27 @@ func newQueryBlock(b *block.Block) queryBlock {
...
 
-	//TODO: Read reward
-	/*
 		if tx.Type() == core.Distribute {
-			distributeTx, _ := tx.Decode()
+			distribute, _ := tx.Decode()
+
+			for _, note := range distribute.Notes {
+				var r uint32
+				if err := encoding.ReadUint32LE(bytes.NewBuffer(note.ValueCommitment), &r); err != nil {
```
[the remark this answers is not in the document]
Comment (goshawk-3) [reply pkg/gql/query/blocks.go:81]: I'm a bit confused. Then why do you marshal it as 32-length:
```golang
func MarshalNote(r *bytes.Buffer, f *Note) error {
	if err := encoding.WriteUint8(r, f.Type); err != nil {
		return err
	}

	if err := encoding.Write256(r, f.ValueCommitment); err != nil {
		return err
	}
```

When this was raised with the team, it was said that we need `ValueCommitment`. So I'll file another issue to address this.

File: pkg/core/mempool/mempool.go [hunk 2]
```diff
@@ -272,15 +272,14 @@ func (m *Mempool) processTx(t TxDesc) ([]byte, error) {
...
 	var hash []byte
 	var err error
 
-	if hash, fee, err = m.verifier.Preverify(ctx, t.tx); err != nil {
+	if hash, _, err = m.verifier.Preverify(ctx, t.tx); err != nil {
 		return nil, err
 	}
 
-	t.tx, err = transactions.Extend(t.tx, fee, hash)
+	t.tx, err = transactions.UpdateHash(t.tx, hash)
```
[the remark this answers is not in the document]
Comment (goshawk-3) [reply pkg/core/mempool/mempool.go:282]: Not needed for now, I think

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**